lib/file-dependencies.rb

```diff
@@ -23,11 +23,11 @@ def process_downloads(files, target, tmpdir)
       full_target = file['target'] ? ::File.join(target, file['target']) : target
       download = FileDependencies::File.fetch_file(file['url'], file['sha1'], tmpdir)
       if (res = download.match(/(\S+?)(\.tar\.gz|\.tgz)$/))
-        prefix = res.captures.first.gsub("#{tmpdir}/", '')
         FileDependencies::Archive.untar(download) do |entry|
+          prefix = file['include_tar_prefix'] ? '' : ::File.join(entry.full_name.split(::File::SEPARATOR).first, '')
           next unless FileDependencies::Archive.extract_file?(entry.full_name, file['extract'], file['exclude'], prefix)
           if file['flatten'] == true
-            ::File.join(full_target, entry.full_name.split("/").last)
+            ::File.join(full_target, entry.full_name.split(::File::SEPARATOR).last)
```

Comment:
I didn't see this last time, but you don't need to do the split().last, you can just use `File.basename(entry.full_name)`